packages/core/src/BiconomyProvider.ts

```diff
@@ -0,0 +1,157 @@
+import { BaseValidationModule, ModuleInfo } from "@biconomy/modules";
+import { Bundler, UserOpResponse } from "@biconomy/bundler";
+import { ChainId, Transaction, UserOperation } from "@biconomy/core-types";
+import { BiconomySmartAccountV2, DEFAULT_ENTRYPOINT_ADDRESS, BuildUserOpOptions } from "@biconomy/account";
+import { BiconomyPaymaster, PaymasterMode } from "@biconomy/paymaster";
+import { IBiconomyProvider } from "./interfaces/IBiconomyProvider";
+import { PaymasterDto, SmartAccountProviderInternalOpts, SmartAccountProviderOpts } from "./Types";
+
+/**
+ * @title BiconomyProvider Class
+ * @dev Implements the IBiconomyProvider interface, acting as a wrapper over various Biconomy packages.
+ */
+export class BiconomyProvider implements IBiconomyProvider {
```

Comment:
should it not extend from BaseProvider? we may be missing some provider properties